OpenSim/Tools/MarkerPlacer.cpp

```diff
@@ -391,23 +392,27 @@ bool MarkerPlacer::processModel(Model* aModel,
     //_outputStorage->print("statesReporterOutputWithMarkers.sto");
 
     if(_printResultFiles) {
-        if (!_outputModelFileNameProp.getValueIsDefault())
-        {
+        std::string savedCwd = IO::getCwd();
+        IO::chDir(aPathToSubject);
+
+        if (_outputModelFileNameProp.isValidFileName()) {
             aModel->print(aPathToSubject + _outputModelFileName);
-            cout << "Wrote model file " << _outputModelFileName << " from model " << aModel->getName() << endl;
+            cout << "Wrote model file " << _outputModelFileName <<
```

Comment:
Could `aModel->print()`, `aModel->writeMarkerFile()`, etc. throw exceptions (e.g., if write access is denied for the specified directory, insufficient space, etc.)? If so, perhaps any exceptions need to be caught here so that the directory can be restored (i.e., `IO::chDir(savedCwd);`), and then re-throw the original exception?

Reply:
That's a good point. I think that is a safe thing to add. I'll add that here and in `ModelScaler`.